scalapb-runtime-grpc/src/main/scala/scalapb/grpc/Marshaller.scala

```diff
@@ -6,10 +6,12 @@ import scalapb.{GeneratedMessage, GeneratedMessageCompanion, TypeMapper}
 
 class Marshaller[T <: GeneratedMessage](companion: GeneratedMessageCompanion[T])
     extends io.grpc.MethodDescriptor.Marshaller[T] {
-  override def stream(t: T): InputStream = new ByteArrayInputStream(t.toByteArray)
+  override def stream(t: T): InputStream = new ProtoInputStream[T](t, this)
 
-  override def parse(inputStream: InputStream): T =
-    companion.parseFrom(inputStream)
+  override def parse(inputStream: InputStream): T = inputStream match {
+    case pis: ProtoInputStream[T] if pis.marshaller == this => pis.message
```

Comment:
what's the reason for comparing the marshallers? if not needed, the `val marshaller` can be removed from the ProtoInputStream class.

Reply:
removed in https://github.com/scalapb/ScalaPB/pull/1615/commits/8d8256cffac1110abd44dcb75727c165c4d4ffbb

introduced it to reproduce the [following precaution logic](https://github.com/grpc/grpc-java/blob/master/protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java#L175-L178), however seems to be redundant indeed.